lib/client.js

```diff
@@ -9,11 +17,375 @@ var dgram = require('dgram'),
+        if (isMetricAggregated) {
+            if (targetBuffer[agg][aggFreq].buffer.length > 0) {
+                targetBuffer[agg][aggFreq].buffer += '\n';
+            }
+
+            targetBuffer[agg][aggFreq].buffer += metricLines;
+            targetBuffer.bufferSize++;
+        } else {
+            if (targetBuffer.bufferSize > 0) {
+                targetBuffer.buffer += '\n';
+            }
+
+            targetBuffer.buffer += metricLines;
+            targetBuffer.bufferSize++;
+        }
+
+
+        if ( (self.aggregatedBuffer.bufferSize + self.nonAggregatedBuffer.bufferSize) >= self.flushSize) {
+            flush(self);
+        }
+    } else {
+        if (self.logger) {
+            self.logger.error('Invalid metric lines: ' + metricLines);
+        }
+    }
+}
+
+/**
+ * Adds a new metric to the in-memory buffer.
+ *
+ * @param self A self client instance.
+ * @param metric Name metric such as 'response_time'.
+ * @param value.
+ * @param parameters An object with metric para meters: tags, agg, aggFreq, namespace and timestamp.
+ *          - tags: Tags to associate this value with, for example {from: 'serviceA', to: 'serviceB', method: 'login'}.
+ *          - agg: List of aggregations to be applied by Statful. Ex: ['avg', 'p90', 'min'].
+ *          - aggFreq: Aggregation frequency in seconds. One of: 10, 30, 60 ,120, 180, 300. Default: 10.
+ *          - namespace: Define the metric namespace. Default: application.
+ *          - timestamp: Defines the metrics timestamp. Default: current timestamp.
+ * @param isMetricAggregated
+ */
+function putRaw(self, metric, value, parameters, isMetricAggregated) {
+    var metricParams = parameters || {};
+
+    var tags = metricParams.tags,
+        agg = metricParams.agg,
+        aggFreq = metricParams.aggFreq,
+        namespace = metricParams.namespace,
+        timestamp = metricParams.timestamp,
+        sampleRate = self.sampleRate;
+
+    // Vars to Put
+    var putNamespace = namespace || self.namespace;
+    var putAggFreq = aggFreq || 10;
+    var putTags = merge(self.app ? merge({app: self.app}, tags) : tags, self.tags);
+
+    var metricName = putNamespace + '.' + metric,
+        flushLine = metricName,
+        sampleRateNormalized = (sampleRate || 100) / 100;
+
+    if ( (Math.random() <= sampleRateNormalized) || isMetricAggregated) {
+        flushLine = Object.keys(putTags).reduce(function (previousValue, tag) {
+            return previousValue + ',' + tag + '=' + putTags[tag];
+        }, flushLine);
+
+        flushLine += ' ' + value + ' ' + (timestamp || (Math.round(new Date().getTime() / 1000)));
+
+        if (agg && !isMetricAggregated) {
+            agg.push(putAggFreq);
+            flushLine += ' ' + agg.join(',');
+
+            if (sampleRate && sampleRate < 100) {
+                flushLine += ' ' + sampleRate;
+            }
+        }
+
+        if (!isMetricAggregated) {
+            addToBuffer(self, [flushLine], isMetricAggregated, null, null);
```

Comment:
This method signature could potentially receive an `aggregationDetails` object that could be checked for undefined and act accordingly in the method implementation - anyway this is ok for now, but I think it'd make it cleaner.